app/assets/javascripts/components/article_finder/article_finder.jsx

```diff
@@ -52,20 +55,38 @@ const ArticleFinder = createReactClass({
     }
   },
 
-  updateFields(key, value) {
-    return this.props.updateFields(key, value);
+  getQueryVariable() {
+    const query = window.location.search.substring(1);
+    const vars = query.split('&');
+    for (let i = 0; i < vars.length; i += 1) {
+        const pair = vars[i].split('=');
+        this.updateFields(pair[0], pair[1]);
+    }
+    },
+  async updateFields(key, value) {
+    await this.props.updateFields(key, value);
+    if (this.props.search_term.length !== 0) { this.buildURL(); }
   },
 
   toggleFilter() {
     return this.setState({
       showFilters: !this.state.showFilters
     });
   },
-
+  buildURL() {
+    let queryStringUrl = window.location.href.split('?')[0];
+    const search_term = encodeURIComponent(this.props.search_term);
+    const search_type = encodeURIComponent(this.props.search_type);
+    const article_quality = encodeURIComponent(this.props.article_quality);
+    const min_views = encodeURIComponent(this.props.min_views);
```

Comment:
Using `.map()` or `.reduce()` might make this a bit easier to read. Let me know if you need help thinking through how to do this!

Reply:
I think .forEach() would be more favourable as .map() would create another array and iterate , what do you think ?

Reply:
Ultimately, any of the options are fine. You're right to be suspicious of taking up new memory, but I don't think creating this small array will be too bad if it adds clarity.  :)

I think the ideal scenario is one where, if we were to add a new search parameter, we would only need to add a key to one place and the function wouldn't grow that much. Creating an array of strings that represents the keys would likely work with this. Does that make sense?

Reply:
Made the changes! :)